src/detect-dataset.c

```diff
@@ -330,12 +330,16 @@ static int SetupSavePath(const DetectEngineCtx *de_ctx,
     const char *dir = ConfigGetDataDirectory();
     BUG_ON(dir == NULL); // should not be able to fail
     char path[PATH_MAX];
```

Comment:
can be moved below

Reply:
The `char path` you mean? why move it?

Reply:
Because it is unused until the inner block after testing PathIsAbsolute